aiosmtpd/smtp.py

```diff
@@ -74,8 +89,16 @@ def __init__(self, handler,
         self.require_starttls = tls_context and require_starttls
         self._tls_handshake_failed = False
         self._tls_protocol = None
+        self.session = None  # type: Session
+        self.message = None  # type: Envelope
         self.transport = None
 
+    def _create_session(self):
+        return Session()
+
+    def _create_envelope(self):
+        return Envelope()
+
```

Comment:
I have no idea how wrong order of commands can cause DOS attack, even if it will rise attribute errors. And you right, there is no way to get envelope or session be none here, and every command checks if it is called in right order.